{
  testTryCreateResource_commitNotCalled();
  try {
    genCtx.finish(mockLogger);
  }
 catch (  UnableToCompleteException e) {
    fail("finish() failed; it should support safely being called any number of times");
  }
}
